CMake/HighFiveConfig.cmake.in

```diff
@@ -1,16 +1,42 @@
 
 # Get HighFive targets
-if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
-  include("${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/HighFiveTargets.cmake")
+
+
+# Combined HighFive
+add_library(HighFive INTERFACE)
+target_compile_definitions(HighFive INTERFACE MPI_NO_CPPBIND)  # No c++ bindings
+
+# Ensure we activate required C++ std
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+  if(CMAKE_VERSION VERSION_LESS 3.8)
+    message(WARNING "HighFive requires minimum C++11. (C++14 for XTensor) \
+        You may need to set CMAKE_CXX_STANDARD in you project")
+  else()
+    # A client request for a higher std overrides this
+    target_compile_features(HighFive INTERFACE cxx_std_11)
+  endif()
 endif()
 
 # If the user sets this flag, all dependencies are preserved.
 # Useful in central deployments where dependencies are not prepared later
 option(HIGHFIVE_USE_INSTALL_DEPS "Use original Highfive dependencies" @HIGHFIVE_USE_INSTALL_DEPS@)
 if(HIGHFIVE_USE_INSTALL_DEPS)
+  # If enabled in the deploy config, request c++14
+  if(@HIGHFIVE_USE_XTENSOR@ AND NOT CMAKE_VERSION VERSION_LESS 3.8)
+    target_compile_features(HighFive INTERFACE cxx_std_14)
+  endif()
+  message("HIGHFIVE @PROJECT_VERSION@: Using original dependencies (HIGHFIVE_USE_INSTALL_DEPS=YES)")
+  target_link_libraries(HighFive INTERFACE HighFive_libheaders HighFive_libdeps)
```

Comment:
It could be elegant to have a few targets. E.g.:

* `HighFive`: bundles default.
* `HighFive_header`: HighFive's include path and C++ version.
* `HighFive_HDF5`: All HDF5 settings.
* `HighFive_HDF5_MPI`: All HDF5 settings with MPI enabled.
* `HighFive_Boost`: Enable Boost, and set include path.
* `HighFive_Eigen`: Enable Eigen, and set include path.
* `HighFive_xtensor`: Enable xtensor, and set include path.

Then the user can mix and match all it wants, if it does not want the default.

Reply:
Now when the user doesnt want the deployed dependencies (default) he's given options to enable/disable Boost, Eigen and xtensor individually. The selected packages then become part of the HighFive interface. I find this to be sufficiently flexible.
Your suggestion is a different alternative; maybe I'm biased since I implemented this solution, but my feeling is that HighFive shouldn't be advertising third party targets, I honestly haven't seen any project doing it, and even though I guess you don't mean to export the original targets, it might be misleading/confusing to the user.